src/platform/GenericOTARequestorStorage.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "GenericOTARequestorStorage.h"
+
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/CHIPTLV.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <platform/OTARequestorInterface.h>
+
+#include <limits>
+
+namespace chip {
+namespace DeviceLayer {
+
+CHIP_ERROR GenericOTARequestorStorage::StoreProviderLocation(const ProviderLocationType & provider)
+{
+    uint8_t buffer[kMaxSerializedSize];
+    TLV::TLVWriter writer;
+
+    writer.Init(buffer);
+    ReturnErrorOnFailure(provider.EncodeForRead(writer, TLV::AnonymousTag(), provider.fabricIndex));
```

Comment:
Is the norm to encode all data in TLV, before storing in the KVS?

Reply:
Looking at `GroupDataProviderImpl` etc. it looks like it is. I think it's to support some level of data versioning. It will be easier to make sure that the data will be read correctly in future firmware versions.